File: src/main/java/org/bakkes/game/view/components/APositionable.java
package org.bakkes.game.view.components;

import org.bakkes.game.view.IPositionable;
import org.newdawn.slick.geom.Vector2f;

public abstract class APositionable implements IPositionable{

    private final Vector2f position = new Vector2f();

    @Override
	public Vector2f getPosition(){
    	return this.position;
    }
}

File: src/main/scala/org/bakkes/game/view/components/APositionable.scala
package org.bakkes.game.view.components;

import org.bakkes.game.model.Position;
import org.newdawn.slick.geom.Vector2f;

abstract class APositionable extends Position {
	private val position:Vector2f  = new Vector2f()
	override def getPosition(): Vector2f = position
}
